### PR TITLE
Removed functionality from deprecated tasks DownloadPackageV0, NuGetInstallerV0, and NuGetRestoreV1

### PR DESCRIPTION
**Pull request by amp-powell**
Removed functionality from deprecated tasks DownloadPackageV0, NuGetInstallerV0, and NuGetRestoreV1

**Task name**: DownloadPackageV0, NuGetInstallerV0, and NuGetRestoreV1

**Description**: Took away functionality so that these deprecated tasks can no longer be used in pipelines

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) Y

**Attached related issue:** (Y/N) Y - User Story 2182164

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/NuGetInstallerV0/Tests/L0.ts [hunk 1]
```diff
@@ -13,13 +13,13 @@ describe('NuGetInstaller Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run()
-        assert(tr.ran('c:\\from\\tool\\installer\\nuget.exe restore -NonInteractive c:\\agent\\home\\directory\\single.sln'), 'it should have run NuGet');
-        assert(tr.stdOutContained('setting console code page'), 'it should have run chcp');
-        assert(tr.stdOutContained('NuGet output here'), "should have nuget output");
-        assert(tr.invokedToolCount == 1, 'should have run NuGet');
+        assert(!tr.ran('c:\\from\\tool\\installer\\nuget.exe restore -NonInteractive c:\\agent\\home\\directory\\single.sln'), 'it should not have run NuGet');
```
[the remark this answers is not in the document]
Comment (amp-powell) [reply Tasks/NuGetInstallerV0/Tests/L0.ts:16]: I changed all of the tests to make sure that the task doesn't run at all now, but I can simplify this to one test if that's better

File: Tasks/DownloadPackageV0/download.ts [hunk 2]
```diff
@@ -17,51 +17,7 @@ import { WebApi } from 'azure-devops-node-api';
...
-
-function getAuthToken() {
-	var auth = tl.getEndpointAuthorization('SYSTEMVSSCONNECTION', false);
-	if (auth.scheme.toLowerCase() === 'oauth') {
-		return auth.parameters['AccessToken'];
-	}
-	else {
-		throw new Error(tl.loc("CredentialsNotFound"))
-	}
+	throw new Error(tl.loc("DeprecatedTask"));
```
[the remark this answers is not in the document]
Comment (MicroTuld) [reply Tasks/DownloadPackageV0/download.ts:20]: Seconded

File: Tasks/DownloadPackageV0/download.ts
Comment (MicroTuld) [inline Tasks/DownloadPackageV0/download.ts:1]: Here and in the other .ts files - if they have been almost completely gutted of functions and only need to return a failed task, how much can we trim from the import clauses?

File: _generated/NuGetRestoreV1_Node20/nugetinstaller.ts [hunk 3]
```diff
@@ -18,294 +18,8 @@ import * as telemetry from "azure-pipelines-tasks-utility-common/telemetry";
...
-            "isIncludeNuGetOrgEnabled": tl.getBoolInput("includeNuGetOrg", false)
-        };
-        telemetry.emitTelemetry("Packaging", "NuGetRestore", telem);
-    } catch (err) {
-        tl.debug(`Unable to log NuGet Tool Installer task init telemetry. Err:(${err})`);
-    }
+    tl.setResult(tl.TaskResult.Failed, tl.loc("DeprecatedTask"));
 }
 
 main();
```
Comment (MicroTuld) [inline _generated/NuGetRestoreV1_Node20/nugetinstaller.ts:25]: What does `main();` do here?

File: _generated/NuGetRestoreV1_Node20/nugetinstaller.ts [hunk 4]
```diff
@@ -18,294 +18,8 @@ import * as telemetry from "azure-pipelines-tasks-utility-common/telemetry";
 const NUGET_ORG_V2_URL: string = "https://www.nuget.org/api/v2/";
 const NUGET_ORG_V3_URL: string = "https://api.nuget.org/v3/index.json";
```
Comment (MicroTuld) [inline _generated/NuGetRestoreV1_Node20/nugetinstaller.ts:19]: Unused `const`s and imports?

File: Tasks/NuGetInstallerV0/Tests/L0.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (amp-powell) [reply Tasks/NuGetInstallerV0/Tests/L0.ts]: I ended up taking away most tests, and just left a single one where I checked for the deprecation message

File: Tasks/DownloadPackageV0/download.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (amp-powell) [reply Tasks/DownloadPackageV0/download.ts]: I was following the format used before, but I changed it now to make it simpler

File: Tasks/DownloadPackageV0/download.ts [hunk 5]
```diff
@@ -17,51 +17,7 @@ import { WebApi } from 'azure-devops-node-api';
...
 }
 
 export async function downloadPackage(feedConnection: WebApi, pkgsConnection: WebApi, feedId: string, packageId: string, version: string, downloadPath: string) {
```
[the remark this answers is not in the document]
Comment (amp-powell) [reply Tasks/DownloadPackageV0/download.ts]: Took them out! I gutted a lot of the task this time

File: _generated/NuGetRestoreV1_Node20/nugetinstaller.ts (hunk 4 above)
Comment (amp-powell) [reply _generated/NuGetRestoreV1_Node20/nugetinstaller.ts]: Removed these since they're no longer needed!

**Review by MicroTuld — APPROVED** [review]
LGTM!

**Review by MicroTuld — APPROVED** [review]
LGTM!

**Pull request merged**